doc/paper/paper.tex

```diff
@@ -0,0 +1,1012 @@
+    data retention.
+
+    \textbf{Support for distributed compute.} Typical server log databases do not
+    expect you to run distributed compute jobs on your log data and are not
+    designed with that use-case in mind, whereas in multimodal log management
+    it is a core workload.
+
+    \textbf{Support for statistical aggregates.} Multimodal logs are logs, but
+    in a lot of cases they are also similar to metrics. For some kinds of logs
+    it is attractive to be able to plot statistical aggregates such as
+    quantiles or averages of field values over selected windows of time. The
+    APM portions of Datadog enable this kind of operation at low latency over
+    trillions or quadrillions of points. To drive a responsive interface these
+    aggregations must rely on precomputation.
+
+    Of course, it can be assumed that in some large organizations these concerns
+    all apply to server logs as well. Many organizations employ more infrastructure
+    engineers than all the employees in the largest robotics company, and some of
+    them undoubtedly produce petabytes of logs and analyze them with Kafka
+    \cite{kafka} or Spark \cite{spark}.
+
+    However, in multimodal logging these scales are hit by startup companies, and
+    the scaling requirements are significantly a function of the product’s
+    properties and requirements – i.e what kinds of sensors and how good the
+    performance needs to be – while in contrast the scaling requirements of
+    traditional log databases are generally driven by organizational scale – i.e
+    how many thousands of internal microservices there are. Organizational scale is
+    slower-growing because it usually requires some form of sustained business
+    success over years.
+
+    In terms of system mechanics, there are a few other common considerations to
+    note in multimodal log data management and modeling. The data comes into the
+    system in a multiplexed format, where inputs generally contain data from
+    multiple topics comingled. This is the most efficient format for writers to
+    write, which is usually the main consideration on the device.
+
+    Within that mixed message stream there is a large amount of variation in size
+    of individual messages, ranging from a few bytes to dozens of megabytes. In
+    addition, the aggregate record volume for these applications can be very
+    large - trillions or quadrillions of records.
+
+    Due to the volumes, a system that stores compressed data files in S3-like
+    storage may be the only practical solution, and in such storage systems there
+    is no ability to do real random IO (only range requests). It is therefore
+    important to structure the writes to storage so that readers can do as much
+    sequential IO as possible. Since the data is accessed by topic, this means
+    storing data in files split or partitioned by topic - unlike the input
+    format.
```

Comment:
Overall, this section seems to have presented a number of different ideas. I wonder if all this background is necessarily needed? If not, perhaps we can consider shortening this section a bit for clarity. Just a suggestion!